core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java

```diff
@@ -238,9 +240,11 @@ public InteractiveTransactionResponse deposit(Sep10Jwt token, Map<String, String
     }
 
     // Validate amount
-    if (strAmount != null) {
-      if (decimal(strAmount).compareTo(decimal(asset.getDeposit().getMinAmount())) < 0
-          || decimal(strAmount).compareTo(decimal(asset.getDeposit().getMaxAmount())) > 0) {
+    Long minAmount = asset.getWithdraw().getMinAmount();
+    Long maxAmount = asset.getWithdraw().getMaxAmount();
+    if (strAmount != null && minAmount != null && maxAmount != null) {
```

Comment:
I think we may start adding comments for the logics that can be confusing. 
```suggestion
    // check the amount only when the min and max amounts are defined.
    if (strAmount != null && minAmount != null && maxAmount != null) {
```

Reply:
I splitted the code to handle min and max separated, so it is less confusing now.